source/app/service-providers/commands/dir-new-project.ts

```diff
@@ -22,11 +22,12 @@ export default class DirNewProject extends ZettlrCommand {
 
   /**
     * Create a new project for a directory.
-    * @param {String} evt The event name
-    * @param  {Object} arg The hash of a directory.
+    *
+    * @param  {string}  evt  The event name
+    * @param  {any}     arg  The payload
     */
-  async run (evt: string, arg: any): Promise<void> {
-    let dir = this._app.workspaces.findDir(arg.path)
+  async run (evt: string, arg: { path: string }): Promise<void> {
+    const dir = await this._app.fsal.getAnyDirectoryDescriptor(arg.path)
```

Comment:
Most changes in the commands refer to moving from workspaces to the FSAL. This also makes the logic simpler in some places. Also, a lot of proper typing in the arg.